android-components/components/support/utils/src/test/java/mozilla/components/support/utils/WebURLFinderTest.kt

```diff
@@ -108,4 +116,97 @@ class WebURLFinderTest {
+        assertTrue(INVERT(isURLLike("file://C:\\Users\\user\\myfile.html")))
+        assertTrue(isURLLike("http://192.168.255.255"))
+        assertTrue(isURLLike("link.unknown"))
+        assertTrue(isURLLike("3.14.2019"))
+        assertTrue(isURLLike("3-four.14.2019"))
+        assertTrue(isURLLike(" cnn.com "))
+        assertTrue(isURLLike(" cnn.com"))
+        assertTrue(isURLLike("cnn.com "))
+        assertTrue(isURLLike("mozilla.com/~userdir"))
+        assertTrue(isURLLike("my-domain.com"))
+        assertTrue(isURLLike("http://faß.de//"))
+        assertTrue(isURLLike("cnn.cơḿ"))
+        assertTrue(isURLLike("cnn.çơḿ"))
+
+        assertTrue(isURLLike("c-c.com"))
+        assertTrue(isURLLike("c-c-c-c.c-c-c"))
+        assertTrue(isURLLike("c-http://c.com"))
+        assertTrue(isURLLike("about-mozilla:mozilla"))
+        assertTrue(isURLLike("c-http.d-x"))
+        assertTrue(isURLLike("www.c.-"))
+        assertTrue(isURLLike("3-3.3"))
+        assertTrue(isURLLike("www.c-c.-"))
+
+        assertFalse(isURLLike(" -://x.com "))
+        assertFalse(isURLLike("  -x.com"))
+        assertFalse(isURLLike("http://www-.com"))
+        assertFalse(isURLLike("www.c-c-  "))
+        assertFalse(isURLLike("3-3 "))
+
+        val validIPv6Literals = listOf(
+            "[::]",
+            "[::1]",
+            "[1::]",
+            "[1:2:3:4:5:6:7:8]",
+            "[2001:db8::1.2.3.4]",
+            "[::1]:8080",
+        )
+
+        validIPv6Literals.forEach { url ->
+            assertTrue(INVERT(isURLLike(url)))
+            assertTrue(INVERT(isURLLike("$url/")))
+            assertTrue(isURLLike("https://$url"))
+            assertTrue(isURLLike("https://$url/"))
+            assertTrue(isURLLike("https:$url"))
+            assertTrue(isURLLike("http://$url"))
+            assertTrue(isURLLike("http://$url/"))
+            assertTrue(isURLLike("http:$url"))
+        }
+
+        assertFalse(isURLLike("::1"))
+        assertFalse(isURLLike(":::"))
+        assertFalse(isURLLike("[[http://]]"))
+        assertFalse(isURLLike("[[["))
+        assertFalse(isURLLike("[[[:"))
+        assertFalse(isURLLike("[[[:/"))
+        assertFalse(isURLLike("http://]]]"))
+
+        assertTrue(isURLLike("fe80::"))
+        assertTrue(isURLLike("x:["))
+
+        assertTrue(INVERT(isURLLike("[:::")))
+        assertTrue(INVERT(isURLLike("http://[::")))
+        assertTrue(INVERT(isURLLike("http://[::/path")))
+        assertTrue(INVERT(isURLLike("http://[::?query")))
+        assertTrue(INVERT(isURLLike("[[http://banana]]")))
+        assertTrue(INVERT(isURLLike("http://[[[")))
+        assertTrue(INVERT(isURLLike("[[[::")))
+        assertTrue(INVERT(isURLLike("[[[::/")))
+        assertTrue(INVERT(isURLLike("http://[1.2.3]")))
+        assertTrue(INVERT(isURLLike("https://[1:2:3:4:5:6:7]/")))
+        assertTrue(INVERT(isURLLike("https://[1:2:3:4:5:6:7:8:9]/")))
+
+        assertTrue(isURLLike("https://abc--cba.com/"))
+    }
```

Comment:
I'm sorry, I think my comment about adding tests for WebUrlFinderTest was not clear. 

Here is what I had in mind: 
```kotlin 
    @Test
    fun testIPv6LiteralsInsideSentence() {
        val validIPv6Literals = listOf(
            "[::]",
            "[::1]",
            "[1::]",
            "[1:2:3:4:5:6:7:8]",
            "[2001:db8::1.2.3.4]",
            "[::1]:8080",
        ).flatMap { url ->
            listOf(
                "https://$url",
                "https://$url/",
                "https:$url",
                "http://$url",
                "http://$url/",
                "http:$url",
            )
        }

        validIPv6Literals.forEach { url ->
            assertEquals(url, find("random_text $url other_random_text"))
        }
    }
```

The idea would be to make sure the regex is able to find the IPv6 literals inside a sentance containing them. 
What do you think?

Reply:
I've updated isUrlLikeEmulated() to use `find("random_text $it other_random_text")`, so it covers the cases you listed and more.